libsquoosh/src/index.ts

```diff
@@ -251,22 +250,26 @@ class Image {
  */
 class ImagePool {
   public workerPool: WorkerPool<JobMessage, any>;
+  public loadFile: (path: URL) => Promise<ArrayBuffer>;
 
   /**
    * Create a new pool.
-   * @param {number} [threads] - Number of concurrent image processes to run in the pool. Defaults to the number of CPU cores in the system.
+   * @param {(path: URL) => Promise<ArrayBuffer>} [loadFile] - A function that loads a file from a URL.
+   * @param {number} [threads] - Number of concurrent image processes to run in the pool.
    */
-  constructor(threads: number) {
-    this.workerPool = new WorkerPool(threads || cpus().length, __filename);
+  constructor(loadFile: (path: URL) => Promise<ArrayBuffer>, threads: number) {
+    this.loadFile = loadFile;
+    this.workerPool = new WorkerPool(threads, __filename);
   }
 
   /**
    * Ingest an image into the image pool.
-   * @param {FileLike} image - The image or path to the image that should be ingested and decoded.
+   * @param {URL} url - The URL path to the image that should be ingested and decoded.
    * @returns {Image} - A custom class reference to the decoded image.
    */
-  ingestImage(image: FileLike): Image {
-    return new Image(this.workerPool, image);
+  ingestImage(url: URL): Image {
+    const file = await this.loadFile(url);
+    return new Image(this.workerPool, file);
```

Comment:
We can't use `await` here unless we make `ingestImage` async as well.

@surma Thoughts?

Reply:
I think that’s fine!

Reply:
But we should still allow buffers to be passed to `ingestImage` as well.

Reply:
Oh in that case, let's just use `ArrayBuffer` as the parameter here.

No need for a special `loadFile()` function since the caller can call do `ingestImage(loadFile())` 🙂  

I just pushed a change that reduced a lot of Node.js specific code that gets us closer to running on the web 👍